File: src/main/java/com/chtl/ast/node/CommentNode.java
package com.chtl.ast.node;

import com.chtl.ast.*;

/**
 * 注释节点
 * 对应CHTL中的三种注释：
 * 1. 行注释 (双斜线)
 * 2. 块注释 (斜线星号)
 * 3. 生成器识别的注释 (双横线)
 */
public class CommentNode extends AbstractCHTLASTNode {
    
    public enum CommentType {
        LINE_COMMENT,        // //
        BLOCK_COMMENT,       // /* */
        GENERATOR_COMMENT    // --
    }
    
    private CommentType commentType;
    private String content;
    
    public CommentNode(CommentType commentType, String content) {
        super(NodeType.COMMENT);
        this.commentType = commentType;
        this.content = content;
    }
    
    public CommentType getCommentType() {
        return commentType;
    }
    
    public void setCommentType(CommentType commentType) {
        this.commentType = commentType;
    }
    
    public String getContent() {
        return content;
    }
    
    public void setContent(String content) {
        this.content = content;
    }
    
    /**
     * 判断是否是生成器注释
     */
    public boolean isGeneratorComment() {
        return commentType == CommentType.GENERATOR_COMMENT;
    }
    
    @Override
    public void accept(CHTLASTVisitor visitor) {
        visitor.visitComment(this);
    }
}

File: CHTLCompiler/src/ast/CommentNode.cpp
#include "ast/CommentNode.h"
#include <sstream>

namespace chtl {

CommentNode::CommentNode(const std::string& comment_content, ASTNodeType type)
    : ASTNode(type), content(comment_content), comment_type(type) {
    generateNodeId();
    node_id = "comment_" + node_id;
}

std::string CommentNode::getContent() const {
    return content;
}

void CommentNode::setContent(const std::string& comment_content) {
    content = comment_content;
}

ASTNodeType CommentNode::getCommentType() const {
    return comment_type;
}

void CommentNode::setCommentType(ASTNodeType type) {
    comment_type = type;
    this->type = type;
}

std::unique_ptr<ASTNode> CommentNode::clone() const {
    auto cloned_node = std::make_unique<CommentNode>(content, comment_type);
    cloned_node->line = line;
    cloned_node->column = column;
    cloned_node->start_position = start_position;
    cloned_node->end_position = end_position;
    cloned_node->source_file = source_file;
    return cloned_node;
}

std::string CommentNode::getNodeTypeName() const {
    switch (comment_type) {
        case ASTNodeType::COMMENT_SINGLE_LINE:
            return "CommentSingleLine";
        case ASTNodeType::COMMENT_MULTI_LINE:
            return "CommentMultiLine";
        case ASTNodeType::COMMENT_GENERATOR:
            return "CommentGenerator";
        default:
            return "Comment";
    }
}

std::string CommentNode::serialize() const {
    std::ostringstream oss;
    oss << "CommentNode {";
    oss << " type: " << getNodeTypeName();
    oss << ", content: \"" << content << "\"";
    oss << ", line: " << line;
    oss << ", column: " << column;
    oss << " }";
    return oss.str();
}

std::vector<std::string> CommentNode::validate() const {
    std::vector<std::string> errors;
    if (content.empty()) {
        errors.push_back("注释内容不能为空");
    }
    if (type != comment_type) {
        errors.push_back("注释类型不匹配");
    }
    return errors;
}

} // namespace chtl